### ISSUE TITLE
Url scheme not allowed error while downloading pango on Arch Linux

### ISSUE THREAD

**Issue by michardy** [kpcyrd/pacman-bintrans#1]
While installing pango I got an error trying to download the transparency proof.

Steps taken:
1. I installed pacman-bintrans 0.1.0-1 from the arch community repository
2. Installed the custom transport by copying the line from the README
3. Ran `pacman -Scc`
4. Ran `pacman -Syu` triggering an update to multiple packages including `libibus-1.5.25-2` and `pango-1:1.48.9-2`

Results:
The libibus download was successfully verified while the pango download failed with the following error:
```
[2021-08-24T02:31:22Z INFO  pacman_bintrans] Writing pkg to "/var/cache/pacman/pkg/libibus-1.5.25-2-x86_64.pkg.tar.zst.part"
[2021-08-24T02:31:22Z INFO  pacman_bintrans] Transparency proof is required for "http://mirrors.sonic.net/archlinux/extra/os/x86_64/pango-1:1.48.9-2-x86_64.pkg.tar.zst", downloading into memory
[2021-08-24T02:31:22Z INFO  pacman_bintrans::proof] Trying to download transparency proof from "pango-1:1.48.9-2-x86_64.pkg.tar.zst.t"
Error: Failed to send request

Caused by:
    0: builder error for url (pango-1:1.48.9-2-x86_64.pkg.tar.zst.t): URL scheme is not allowed
    1: URL scheme is not allowed
```

**Comment by kpcyrd** [comment]
hi, thanks for reporting this!
```
[2021-08-24T02:31:22Z INFO  pacman_bintrans::proof] Trying to download transparency proof from "pango-1:1.48.9-2-x86_64.pkg.tar.zst.t"
```
This should be a url, can you share the command you're using?

**Comment by kpcyrd** [comment]
`Url::join` was behaving unexpected with inputs containing a `:`, basically parsing it as `pango-1://1.48.9-2-x86_64.pkg.tar.zst.t`. This was fixed in a 0.1.1 release, thanks!

**Comment by michardy** [comment]
Thanks that update fixed the bug on my end.
